src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -150,12 +150,28 @@ public override void DeleteFile(string fullPath)
             if (Interop.Sys.Unlink(fullPath) < 0)
             {
                 Interop.ErrorInfo errorInfo = Interop.Sys.GetLastErrorInfo();
-                // ENOENT means it already doesn't exist; nop
-                if (errorInfo.Error != Interop.Error.ENOENT)
+                switch(errorInfo.Error)
                 {
-                    if (errorInfo.Error == Interop.Error.EISDIR)
+                    case Interop.Error.ENOENT:
+                        // ENOENT means it already doesn't exist; nop
+                        return;
+                    case Interop.Error.EROFS:
+                        // EROFS means the file system is read-only
+                        // Need to manually check file existence
+                        // github.com/dotnet/corefx/issues/21273
```

Comment:
I usually agree with the sentiment- this particular one has lots of context that I would prefer to have more readily available to help discourage regressions. It is actually a fixed bug in the Linux kernel- I suspect that we'll be revisiting this a few times.